http/obmc_ssh.hpp

```diff
@@ -0,0 +1,228 @@
+    }
+
+    void doRead()
+    {
+        pipeOut.async_read_some(
+            boost::asio::buffer(outputBuffer.data(), outputBuffer.size()),
+            [this, self(shared_from_this())](
+                const boost::system::error_code& ec, std::size_t bytesRead) {
+                BMCWEB_LOG_DEBUG << "Read done.  Read " << bytesRead
+                                 << " bytes";
+                if (ec)
+                {
+                    BMCWEB_LOG_ERROR << "Couldn't read from ssh port: " << ec;
+                    if (session != nullptr)
+                    {
+                        session->close("Error in connecting to ssh port");
+                    }
+                    return;
+                }
+                if (session == nullptr)
+                {
+                    return;
+                }
+
+                if (!outputBuffer.empty())
+                {
+                    std::string_view payload1(cRLineFeed);
+                    session->sendBinary(payload1);
+                    std::string output(outputBuffer.data(), bytesRead);
+                    size_t pos = output.find(lineFeed);
+                    // Repeat till end is reached
+                    while (pos != std::string::npos)
+                    {
+                        // Replace this occurrence of Sub String
+                        output.replace(pos, strlen(lineFeed), cRLineFeed);
+                        // Get the next occurrence from the current position
+                        pos = output.find(lineFeed, pos + strlen(cRLineFeed));
+                    }
+                    std::string_view payload(output.data(), output.size());
+                    session->sendBinary(payload);
+                }
+
+                std::string_view payload(xtermNewLine);
+                session->sendBinary(payload);
+                doRead();
+            });
+    }
+
+    boost::process::async_pipe pipeOut;
+    boost::process::async_pipe pipeIn;
+    boost::process::child processSSH;
+    bool doingWrite;
+};
+
+static std::shared_ptr<Handler> handler;
+
+inline void requestRoutes(App& app)
+{
+    BMCWEB_ROUTE(app, "/bmc-console")
+        .privileges({{"ConfigureManager"}})
+        .websocket()
+        .onopen([](crow::websocket::Connection& conn,
+                   const std::shared_ptr<bmcweb::AsyncResp>&) {
+            BMCWEB_LOG_DEBUG << "Connection " << &conn << " opened";
+
+            if (session != nullptr)
+            {
+                conn.close("Session already connected");
+                return;
+            }
+
+            if (handler != nullptr)
+            {
+                conn.close("Handler already running");
+                return;
+            }
+
+            session = &conn;
+
+            handler = std::make_shared<Handler>(conn.getIoContext());
+            handler->connect();
+        })
+        .onclose([](crow::websocket::Connection& conn,
+                    const std::string& /*reason*/) {
```

Comment:
Should we log something here? 
Make this easier to debug?
https://github.com/openbmc/bmcweb/blob/9a128eb3fc3008f802974360e98de6495ed1a66b/include/nbd_proxy.hpp#L423